detections/endpoint/multiple_users_attempting_to_authenticate_using_explicit_credentials.yml

```diff
@@ -0,0 +1,66 @@
+name: Multiple Users Attempting To Authenticate Using Explicit Credentials
+id: e61918fa-9ca4-11eb-836c-acde48001122
+version: 1
+date: '2021-04-13'
+author: Mauricio Velazco, Splunk
+type: batch
+datamodel:
+- Endpoint
+description: 'The following analytic identifies a source user failing to authenticate
+  with multiple users using explicit credentials on a host. This behavior could represent
+  an adversary performing a Password Spraying attack against an Active Directory environment
+  to obtain initial access or elevate privileges. In this logic, the number of unique
+  users is hardcoded to `10` but can/should be customized to fit especific environments.
+  Event 4648 is generated when a process attempts an account logon by explicitly specifying
+  that account`s credentials. This event generates on domain controllers, member servers,
+  and workstations.\
+
+  This detection will trigger on the potenfially malicious host, perhaps controlled
+  via a trojan or operated by an insider threat, from where a password spraying attack
+  is being executed.\
+
+  The analytic`s returned fields allow analysts to investigate the event further by
+  providing fields like source account, attempted user accounts and the endpoint were
+  the behavior was identified.'
+search: ' `wineventlog_security` EventCode=4648 | bucket span=2m _time | eval Source_Account
```

Comment:
Mauricio, this one would definitely need slots longer than 2 minutes ... see dev-rz, for SV5-ITO-OKTA001.splunkcorp.com which is a global auth server, and source account srv-ito-okta (in one slot it authenticates 9 accounts, then 21, then 12 .... until avg and stdev stabilize (if at all) it would throw many unnecessary detections.

`sourcetype="WinEventLog" OR sourcetype="wineventlog" index="*"EventCode=4648 | bucket span=2m _time | eval Source_Account
  = mvindex(Account_Name, 0) | eval Destination_Account = mvindex(Account_Name, 1)
  | search Source_Account != "*$" Source_Account !="-" Destination_Account !="*$"
  | stats dc(Destination_Account) AS unique_accounts values(Destination_Account) as
  tried_account by _time, ComputerName, Source_Account | eventstats count as numSlots by ComputerName, Source_Account | where numSlots > 1`

Reply:
Misko, I think this is expected. Enterprise environments will always have multi-user systems where several users will be authenticating (or failing to authenticate) from one source. These include but are not limited to VPN terminators, Identity providers, Exchange servers, Web applications like an Intranet, Citrix farms, etc.  

On the false positive documentation for this detection, I clearly describe that this type of systems will have to be identified and added to an allow list (https://github.com/splunk/security_content/blob/password_spraying_TR-515/detections/endpoint/multiple_users_attempting_to_authenticate_using_explicit_credentials.yml#L35 ). This is not anything new for SOCs, its normal to have exceptions for certain detections. In the case of Dev RZ, once you add these servers to an allow list, there is no other systems showing this behavior. 

Finally, since we plan to schedule this query to run every hour, it should only take the first time to allow analysts to identify these systems and whitelist them.